src/implementations/exponential.jl

```diff
@@ -0,0 +1,136 @@
+# Inputs
+# ------
+function copy_input(::typeof(exponential), A::AbstractMatrix)
+    return copy!(similar(A, float(eltype(A))), A)
+end
+
+copy_input(::typeof(exponential), A::Diagonal) = copy(A)
+
+function copy_input(::typeof(exponentiali), τ::Number, A::AbstractMatrix)
+    return τ, copy!(similar(A, complex(eltype(A))), A)
+end
+
+copy_input(::typeof(exponentiali), τ::Number, A::Diagonal) = τ, copy(A)
+
+function check_input(::typeof(exponential!), A::AbstractMatrix, expA::AbstractMatrix, alg::AbstractAlgorithm)
+    m, n = size(A)
+    m == n || throw(DimensionMismatch("square input matrix expected. Got ($m,$n)"))
+    @check_size(expA, (m, m))
+    return @check_scalar(expA, A)
+end
+
+function check_input(::typeof(exponential!), A::AbstractMatrix, expA::AbstractMatrix, alg::MatrixFunctionViaEigh)
+    if !ishermitian(A)
+        throw(DomainError(A, "Hermitian matrix was expected. Use `project_hermitian` to project onto the nearest hermitian matrix)"))
+    end
```

Comment:
TODO: do we want to keep this check here, knowing that it will again be checked in the iimplementation of `eigh_full`?